fern/pages/sdks/features/discriminated-unions.mdx

```diff
@@ -1,33 +1,77 @@
-  ### Discriminated Unions
-  The SDKs natively support discriminated unions. 
+### Discriminated Unions
+The SDKs natively support discriminated unions.
 
-  ```ts maxLines=0 {1}
-  export type Shape = Triangle | Square; 
+<Tabs>
+<Tab title="TypeScript">
 
-  export interface Triangle {
-    type: "triangle";
-    a: number;
-    b: number;
-    c: number;
-  }
+```ts maxLines=0 {1}
+export type Shape = Triangle | Square;
+
+export interface Triangle {
+  type: "triangle";
+  a: number;
+  b: number;
+  c: number;
+}
+
+export interface Square {
+  type: "square";
+  /* length of a single side */
+  side: number;
+}
+```
 
-  export interface Square {
-    type: "square";
-    /* length of a single side */
-    side: number;
+Consumers can easily write branching logic by checking the discriminant.
+
+```ts {4, 6}
+import { Shape } from "sdk";
+
+export function computeArea(shape: Shape): number {
+  if (shape.type === "triangle") {
+    // compute triangle area
+  } else if (shape.type === "square") {
+    // compute square area
   }
-  ```
+}
+```
+
+</Tab>
 
-  Consumers can easily write branching logic by checking the discriminant. 
+<Tab title="Go">
 
-  ```ts {4, 6}
-  import { Shape } from "sdk";
+Go does not have a built-in support for discriminated unions. However, you can define a union struct
+to achieve the same effect:
 
-  export function computeArea(shape: Shape): number {
-    if (shape.type === "triangle") {
+```go maxLines=0 {1-5}
```

Comment:
For go, i'd want to know: 
1) how do i instantiate the generated union 
2) how do i handle it when receive the response